modules/data.atmosphere/tests/testthat/test.download.AmerifluxLBL.R

```diff
@@ -0,0 +1,76 @@
+
+test_that("download respects overwrite argument", {
+	outdir <- withr::local_tempdir()
+	zippath <- file.path(outdir, "AMF_US-Akn_BASE-BADM_6-5.zip")
+	csvpath <- sub("-BADM(.*).zip", "_HH\\1.csv", zippath)
+	
+	# Mock out amerifluxr functions to test our code without network calls
+	local_mocked_bindings(
+		amf_download_base = \(...) {
+			tmp_csv <- basename(csvpath)
+			withr::with_tempdir({
+				writeLines(
+					c(  "# fake file",
+						"#",
+						"TIMESTAMP_START,TIMESTAMP_END",
+						"201101010000,201101010030",
+						"201112310000,201112310030"),
+					tmp_csv)
+				zip(zippath, tmp_csv, flags = "-qr0X")})
+			zippath
+		},
+		amf_var_info = \(...) data.frame(
+			Site_ID = "US-Akn",
+			BASE_Version = "6-5"),
+		.package = "amerifluxr"
```

Comment:
@meetagrawal09 Can you comment in general on whether my mocking approach makes sense here, and specifically check my reasoning for going against the testthat recommendations? 

The testthat docs advise _not_ mocking objects from other namespaces because [my paraphrasings]: (1) it'll affect anything else that calls into that namespace during the life of the mock, even if from outside the code under test, and (2) the point of namespaces is to be sealed, dangit. They recommend instead to either import the functions in question or to encapsulate them in wrapper functions and then mock those. 

For this case both of those options seem like a big extra layer of indirection and I don't see any risk of the mock affecting unintended code. Thoughts?

Reply:
I think that the approach you have taken by mocking these internal functions which take care of the data download is actually the right way to test the emphasis of `overwrite`. Honestly when I write unit tests the motive is to make them run as isolated as possible and mocking functions that deal with a third party data download in one such step towards isolation.